### ISSUE TITLE
Duplicate Enchantment on Tool Using Mana Enchanter

### ISSUE THREAD

**Issue by Shockwve** [Vazkii/Botania#198]
Bug: Duplicate Enchantment on Tool using the Mana Enchanter. I.E. Silk Touch I shows up on tool tip twice.

How to recreate:
1) Put tool on Mana Enchanter
2) Drop Book with Silk Touch I + Sharpness III book
3) Enchant Tool with that book.

**Comment by Arminius** [comment]
Hello, I'd like to open this issue up again. It still exists, but I believe I can shed some light onto what might be causing this - I've done a bit of testing.

It seems to be a cross-mod issue. For the record, I am using the Phoenix Reborn modpack on the FTB Launcher. The following steps reproduce the issue reliably (I used creative mode to avoid expierence costs for these steps):

1. Get an enchanted book of Sharpness V through any means (in my case, I tried both taking it from the creative menu and using a full stack of Nether Quartz in the Enchanter from Ender IO)
2. Get the sword The Ender from Ender IO.
3. Upgrade The Ender in an Anvil using a Vibrant Crystal to get an Empowered The Ender (This doesn't actually add an enchantment to The Ender, and The Ender can still be enchanted normally using a vanilla enchanting table)
4. Enchant The Ender in the Mana Enchanter with the book of Sharpness V

The result is an Empowered The Ender with two copies of the Sharpness V enchant.
If you do not upgrade The Ender before enchanting it, it will only have one enchant as normal (at least it did in my case).

There's probably more things involved here than just Ender IO - I get similar duplication effects on tools and armor from other mods, including those from Botania itself - but I hope this might point you in the right direction.

**Comment by Vazkii** [comment]
Did you reproduce this more than once?

**Comment by Arminius** [comment]
Yes, in fact I just tried it again in the Direwolf20 Pack for 1.7.1 (Pack v1.0.3). The same procedure yields the same results.

**Comment by MrKunji** [comment]
It seems to occur even in 1.7.2. I saw it happen in the 'Magicality: Next Dimension' pack (in a video, a Manasteel Sword got two Knockback I enchants from one book)

**Comment by Vazkii** [comment]
But does it happen every time or only some times? Because I believe I have reproduced it once or twice randomly but never figured out what made it happen.

**Comment by Arminius** [comment]
For me, my procedure works every time.

**Comment by Vazkii** [comment]
That's definitely something I have to experiment with sometime then.

**Comment by TheMadmanofChaos** [comment]
I have looked into this myself, and I may have found the trigger: NBT data. Using MC edit, I added random NBT tags to vanilla diamond swords. In ten tests, all ten swords suffered from the double enchant bug. I then tested the terra steel helm with one of the ancient wills added, to receive the same result.

Looking through the previous comments, many of the items(Such as the ender) to in fact have NBT data added, as he did state he needed to upgrade it in order to achieve the dupe enchantments.
